patroni/ha.py

```diff
@@ -804,6 +804,34 @@ def fetch_nodes_statuses(self, members: List[Member]) -> List[_MemberStatus]:
         pool.join()
         return results
 
+    def fetch_node_status_and_priority(self, member: Member) -> Tuple[_MemberStatus, int]:
+        """This function performs an http get request on member.api_url and fetches its status
+        and priority
+        :returns tuple of `_MemberStatus, int`"""
+
+        try:
+            response = self.patroni.request(member, timeout=2, retries=0)
+            data = response.data.decode('utf-8')
+            logger.info('Got response from %s %s: %s', member.name, member.api_url, data)
+            return (
+                _MemberStatus.from_api_response(member, json.loads(data)),
+                member.failover_priority
```

Comment:
It could be just another field in the `_MemberStatus`.
Also, the priority should be taken from REST API rather than from the member key (what if it was just updated in the YAML file and Patroni didn't update the member key yet?).

Reply:
This makes sense, updated